pkg/network/usm/utils/debugger_testutils.go

```diff
@@ -64,7 +65,7 @@ func WaitForPathToBeBlocked(t *testing.T, programType string, path string) {
 	require.Eventuallyf(t, func() bool {
 		blocked := debugger.GetBlockedPathIDs(programType)
 		for _, id := range blocked {
-			if id == pathID {
+			if id.PathIdentifier == pathID && strings.Contains(id.SamplePath, path) {
```

Comment:
The new condition is problematic and can lead to flakiness
Who guarantees the sample path will match to the given `path`?

Reply:
removed